tests/distributions/test_relaxed_straight_through.py

```diff
@@ -0,0 +1,71 @@
+from __future__ import absolute_import, division, print_function
+
+import torch
+from torch.autograd import grad
+from torch.distributions import OneHotCategorical, RelaxedOneHotCategorical, constraints
+
+import pyro
+import pyro.optim as optim
+import pytest
+from pyro.distributions import RelaxedOneHotCategoricalStraightThrough
+from pyro.infer import SVI, Trace_ELBO
+from tests.common import assert_equal
+
+PROBS = [
+        [0.25, 0.75],
+        [0.25, 0.5, 0.25],
+        [[0.25, 0.75], [0.75, 0.25]],
+        [[[0.25, 0.75]], [[0.75, 0.25]]],
+        [0.1] * 10,
+]
+
+
+@pytest.mark.parametrize('probs', PROBS)
+def test_shapes(probs):
+        temperature = torch.tensor(0.5)
+        probs = torch.tensor(probs, requires_grad=True)
+        d = RelaxedOneHotCategoricalStraightThrough(temperature, probs=probs)
+        sample = d.rsample()
+        log_prob = d.log_prob(sample)
+        grad_probs = grad(log_prob.sum(), [probs])[0]
+        assert grad_probs.shape == probs.shape
+
+
+@pytest.mark.parametrize('temp', [0.2, 0.5, 1.0])
+def test_entropy_grad(temp):
+    num_samples = 1000000
```

Comment:
Why does 1e6 samples only lead to `prec=0.1`? Is this due to finite temperature effects?

Reply:
it's partially that. it's also that the scale isn't 1.0. some of the elements of `expected` are of size 7.6 or what not